File: src/Entity/Reclamation.php
<?php

namespace App\Entity;
use App\Repository\ReclamationRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

#[ORM\Entity(repositoryClass: ReclamationRepository::class)]
class Reclamation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idRec=null;

   
    #[ORM\Column(length: 1000)]
    #[Assert\NotNull(message:"Contenu est obligatoire.")]
    private ?string $contenu=null;

    
    #[ORM\Column(type: Types::DATE_MUTABLE, nullable: true)]
    private ?\DateTimeInterface $dateRec=null;

    
    #[ORM\Column(length: 255)]
    private ?string $etat=null;

    #[ORM\Column(length:2000, nullable: true)]
    private ?string $img=null;

  
    #[ORM\ManyToOne(targetEntity: TypeRec::class, inversedBy: 'reclamations')]
    #[ORM\JoinColumn(name: 'id_type', referencedColumnName: 'id_type')]
    private ?TypeRec $typeRec=null;


    #[ORM\ManyToOne(targetEntity: Utilisateur::class, inversedBy: 'reclamations')]
    #[ORM\JoinColumn(name: 'id_user', referencedColumnName: 'id_user')]
    private ?Utilisateur $idUser=null;


    #[ORM\OneToMany(mappedBy: 'idReclamation', targetEntity: Reponserec::class)]
    private Collection $reponserecs;

    public function __construct()
    {
        $this->reponserecs = new ArrayCollection();

    }

    /**
     * @return Collection<int, Reponserec>
     */
    public function getReponserecs(): Collection
    {
        return $this->reponserecs;
    }

    public function getIdRec(): ?int
    {
        return $this->idRec;
    }

    public function getContenu(): ?string
    {
        return $this->contenu;
    }

    public function setContenu(string $contenu): self
    {
        $this->contenu = $contenu;

        return $this;
    }

    public function getDateRec(): ?\DateTimeInterface
    {
        return $this->dateRec;
    }

    public function setDateRec(\DateTimeInterface $dateRec): self
    {
        $this->dateRec = $dateRec;

        return $this;
    }

    public function getEtat(): ?string
    {
        return $this->etat;
    }

    public function setEtat(string $etat): self
    {
        $this->etat = $etat;

        return $this;
    }

    public function getImg() : ?String
    {
        return $this->img;
    }

    public function setImg(String $img): self
    {
        $this->img = $img;

        return $this;
    }

    public function getTypeRec(): ?TypeRec
    {
        return $this->typeRec;
    }

    public function setTypeRec(?TypeRec $typeRec): self
    {
        $this->typeRec = $typeRec;

        return $this;
    }

    public function getIdUser(): ?Utilisateur
    {
        return $this->idUser;
    }

    public function setIdUser(?Utilisateur $idUser): self
    {
        $this->idUser = $idUser;

        return $this;
    }

    public function addReponserec(Reponserec $reponserec): self
    {
        if (!$this->reponserecs->contains($reponserec)) {
            $this->reponserecs->add($reponserec);
            $reponserec->setIdReclamation($this);
        }

        return $this;
    }

    public function removeReponserec(Reponserec $reponserec): self
    {
        if ($this->reponserecs->removeElement($reponserec)) {
            // set the owning side to null (unless already changed)
            if ($reponserec->getIdReclamation() === $this) {
                $reponserec->setIdReclamation(null);
            }
        }

        return $this;
    }


}

File: src/com/mycompany/myapp/entities/Reclamation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

import java.util.Date;

/**
 *
 * @author Dell 6540
 */
public class Reclamation {
    private int idRec;
    private String contenu;
    private Date dateRec;
    private String etat;
    private String img;
    private int idType;
    private int idUser;
    private int reponserecs;

    public Reclamation() {
    }

    public Reclamation(int idRec, String contenu, Date dateRec, String etat, String img, int idType, int idUser, int reponserecs) {
        this.idRec = idRec;
        this.contenu = contenu;
        this.dateRec = dateRec;
        this.etat = etat;
        this.img = img;
        this.idType = idType;
        this.idUser = idUser;
        this.reponserecs = reponserecs;
    }

    public Reclamation(String contenu, Date dateRec, String etat, String img, int idType, int idUser, int reponserecs) {
        this.contenu = contenu;
        this.dateRec = dateRec;
        this.etat = etat;
        this.img = img;
        this.idType = idType;
        this.idUser = idUser;
        this.reponserecs = reponserecs;
    }

    public int getIdRec() {
        return idRec;
    }

    public void setIdRec(int idRec) {
        this.idRec = idRec;
    }

    public String getContenu() {
        return contenu;
    }

    public void setContenu(String contenu) {
        this.contenu = contenu;
    }

    public Date getDateRec() {
        return dateRec;
    }

    public void setDateRec(Date dateRec) {
        this.dateRec = dateRec;
    }

    public String getEtat() {
        return etat;
    }

    public void setEtat(String etat) {
        this.etat = etat;
    }

    public String getImg() {
        return img;
    }

    public void setImg(String img) {
        this.img = img;
    }

    public int getIdType() {
        return idType;
    }

    public void setIdType(int idType) {
        this.idType = idType;
    }

    public int getIdUser() {
        return idUser;
    }

    public void setIdUser(int idUser) {
        this.idUser = idUser;
    }

    public int getReponserecs() {
        return reponserecs;
    }

    public void setReponserecs(int reponserecs) {
        this.reponserecs = reponserecs;
    }
            
                   
}